patches/BUNDLE-2554__set_payment_info_bug_fix__3.4.1.patch

```diff
@@ -0,0 +1,126 @@
+diff --git a/vendor/amzn/amazon-pay-module/Model/OrderInformationManagement.php b/vendor/amzn/amazon-pay-module/Model/OrderInformationManagement.php
+index 540b357..c18ef91 100644
+--- a/vendor/amzn/amazon-pay-module/Model/OrderInformationManagement.php
++++ b/vendor/amzn/amazon-pay-module/Model/OrderInformationManagement.php
+@@ -144,24 +144,11 @@ public function saveOrderInformation($amazonOrderReferenceId, $allowedConstraint
+             ];
+
+             $responseParser = $this->clientFactory->create($storeId)->setOrderReferenceDetails($data);
+-            try {
+-                $response       = $this->amazonSetOrderDetailsResponseFactory->create(
+-                    [
+-                    'response' => $responseParser
+-                    ]
+-                );
++            $response = $this->amazonSetOrderDetailsResponseFactory->create([
++                'response' => $responseParser
++            ]);
+
+-                $this->validateConstraints($response, $allowedConstraints);
+-            } catch (AmazonServiceUnavailableException $e) {
+-                if($e->getApiErrorCode() == 'OrderReferenceNotModifiable') {
+-                    $this->logger->warning(
+-                        "Could not modify Amazon order details for $amazonOrderReferenceId: "
```

Comment:
Don't embed variables in strings. Use single quotes and concatenate or use sprintf

Reply:
This is a fix for 3rd-party extension, can be ignored